backend/slackbot/handler.js

```diff
@@ -49,190 +50,207 @@ async function handler (event) {
+ * @returns filteredApps The JSON object with appIdStore as the key and name, dates and versions as values
  */
 async function getApps(store='both') {
   const endpoint = gatewayURL + '/apps';
   let filteredApps = {};  // apps we want to return depending on store specified
-  store = store.toLowerCase();
 
-  const appsRequest = await axios.get(endpoint); // Get list of available apps
+  let appsRequest = await axios.get(endpoint); // Get list of available apps
   const apps = appsRequest.data;
 
-  // Filter apps based on if a store is specified
   if (store == 'both') {
     return apps;
   } else {
-
-    for (let key in apps) {
-
-      if (store.includes('google') || store.includes('android')) {
-        if (key.includes('Google Play')) {
-          filteredApps[key] = apps[key];
-        }
+    // Filter apps based on appStoreId and user provided store
+    for (let appIdStore in apps) {
+      if ( (store.includes('google') || store.includes('android')) && appIdStore.includes('Google Play') ) {
+        filteredApps[appIdStore] = apps[appIdStore];
       }
-      else if (store.includes('apple') || store.includes('ios') || store.includes('app')) {
-        if (key.includes('App Store')) {
-          filteredApps[key] = apps[key];
-        }
+      else if ( (store.includes('apple') || store.includes('ios')) && appIdStore.includes('App Store') ) {
+        filteredApps[appIdStore] = apps[appIdStore];
       }
     }
-
   }
 
   return filteredApps;
 }
 
 /**
- * @param apps the list of apps we want statistics from
- * @param parameters any optional or required parameters passed
- * @returns statsList a list of statistics for each specified version of an app
+ * Function takes the apps and slackInput and builds the parameters to request stats for
+ * @param apps The JSON object of all of the apps already filtered by store
+ * @param slackInput The potential user supplied arguments
+ * @returns statsRequests The array of parameters to get stats for
  */
-async function getStatistics(apps, parameters={}){
-  let promises = [];  // Array for promises
-  let statsList = [];
-
-  const endpoint = gatewayURL + '/stats';
+function buildParameters(apps, slackInput) {
+  let statsRequests = [];
 
-  for (let app in apps){
-    let appData = apps[app];
+  for (let app in apps) {
+    let params = {};
 
-    // Using last 7 days
     let days = 7;
-    if (parameters.hasOwnProperty('days')) {
-      days = parameters.days;
+    if (slackInput.hasOwnProperty('days')) {
+      days = slackInput.days;
     }
 
     // Calculate date range using days or defaulting to 7 days back from latest available
     let endDate = new Date();
     let startDate = new Date();
-
     startDate.setDate(endDate.getDate() - days);
-    if (parameters.hasOwnProperty('startDate')) {
-      startDate = new Date(parameters.startDate);
-      startDate = startDate.toISOString();
+
+    if (slackInput.hasOwnProperty('startDate')) {
```

Comment:
Make the two states (startDate/endDate vs. days) into a giant if/else if/else to make it easier to read